client/src/lib/board/socketEmitUtils.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  AddObjectData,
+  AddSlideData,
+  DeleteObjectData,
+  DeleteSlideData,
+  JoinBoardData,
+  JoinSlideData,
+  LeaveBoardData,
+  LeaveSlideData,
+  UpdateObjectData,
+} from "@/interfaces/socket/SocketEmitsData";
+import { Socket } from "socket.io-client";
+
+// Utility function to handle optional callbacks
+const emitWithCallback = (socket: Socket, event: string, data: any, callback?: (res: any) => void) => {
+  if (callback) {
+    socket.emit(event, data, callback);
+  } else {
+    socket.emit(event, data);
+  }
+};
+
+// BOARD //
+export const socketEmitJoinBoard = (socket: Socket, joinBoardData: JoinBoardData, callback?: (res: any) => void) => {
+  emitWithCallback(socket, "join-board", joinBoardData, callback);
+};
+
+export const socketEmitLeaveBoard = (socket: Socket, leaveBoardData: LeaveBoardData, callback?: (res: any) => void) => {
+  emitWithCallback(socket, "leave-board", leaveBoardData, callback);
+};
+
+// SLIDES //
+export const socketEmitAddSlide = (socket: Socket, addSlideData: AddSlideData, callback?: (res: any) => void) => {
+  emitWithCallback(socket, "add-slide", addSlideData, callback);
+};
+
+export const socketEmitDeleteSlide = (
+  socket: Socket,
+  deleteSlideData: DeleteSlideData,
+  callback?: (res: any) => void
+) => {
+  emitWithCallback(socket, "delete-slide", deleteSlideData, callback);
+};
+
+export const socketEmitJoinSlide = (socket: Socket, joinSlideData: JoinSlideData, callback?: (res: any) => void) => {
+  emitWithCallback(socket, "join-slide", joinSlideData, callback);
+};
+
+export const socketEmitLeaveSlide = (socket: Socket, leaveSlideData: LeaveSlideData, callback?: (res: any) => void) => {
```

Comment:
Is this used somewhere?

Reply:
No, this emitter is not used anywhere right now, but I thought it would be convenient to have all the possible emit methods defined in one place. Just in case they are needed somewhere. Additionally, this way we can know which messages are valid ones without having to guess or browse through the backend code.